lamindb/_file.py

```diff
@@ -55,16 +56,15 @@
 
 
 def process_pathlike(
-    filepath: Union[Path, UPath], skip_existence_check: bool = False
+    filepath: UPath, skip_existence_check: bool = False
 ) -> Tuple[Storage, bool]:
     if not skip_existence_check:
         try:  # check if file exists
             if not filepath.exists():
                 raise FileNotFoundError(filepath)
         except PermissionError:
             pass
-    if not isinstance(filepath, UPath):
-        filepath = filepath.resolve()
+    filepath = filepath.resolve()  # works for UPath also
```

Comment:
This isn't needed for a CloudPath object as by definition, the cloudpath is fully specified including the bucket.

Hence: The check for a local path makes sense. Only if it's a local path, we'd like to resolve it. Calling it on CloudPath, too, causes an API request, I fear, and hence unnecessarily reduces performance.

I've made several experiments ingesting high numbers of files and found that reducing the number of API requests is the only factor that determines performance if files aren't big.

Reply:
Agree, will remove it.